vendor/github.com/openshift/installer/pkg/aro/dnsmasq/dnsmasq.go

```diff
@@ -36,9 +36,9 @@ After=network-online.target
 Before=bootkube.service
 
 [Service]
-ExecStartPre=/bin/bash -c '/bin/cp /etc/resolv.conf /etc/resolv.conf.dnsmasq; /bin/sed -ni -e "/^nameserver /!p; \\$$a nameserver $$(hostname -I)" /etc/resolv.conf; /usr/sbin/restorecon /etc/resolv.conf'
+ExecStartPre=/bin/bash -c 'if /bin/grep $(hostname -I|cut -f1 -d" ") /etc/resolv.conf.dnsmasq; then echo "already replaced resolv.conf.dnsmasq"; else /bin/cp /etc/resolv.conf /etc/resolv.conf.dnsmasq; fi; /bin/sed -ni -e "/^nameserver /!p; \\$$a nameserver $$(hostname -I)" /etc/resolv.conf; /usr/sbin/restorecon /etc/resolv.conf'
```

Comment:
@Makdaam 
`/etc/resolv.conf.dnsmasq` should have the upstream DNS server's IP address, not the local node's IP address which it looks like is being tested for in the `grep`.  

It might just make sense to check if the file `/etc/resolv.conf.dnsmasq` exists, since it's being moved on `ExecStopPost`

```suggestion
 ExecStartPre=/bin/bash -c 'if /usr//bin/test -f "/etc/resolv.conf.dnsmasq"; then echo "already replaced resolv.conf.dnsmasq"; else /bin/cp /etc/resolv.conf /etc/resolv.conf.dnsmasq; fi; /bin/sed -ni -e "/^nameserver /!p; \\$$a nameserver $$(hostname -I)" /etc/resolv.conf; /usr/sbin/restorecon /etc/resolv.conf'
```

Reply:
Running this under the assumption dnsmasq was killed you can see what happens below.
```bash
$ cat /etc/resolv.conf
# Generated by NetworkManager
search zkloo21cxfherh0mi15oh2qojb.bx.internal.cloudapp.net
nameserver 10.0.0.7 
$ cat /etc/resolv.conf.dnsmasq 
# Generated by NetworkManager
search zkloo21cxfherh0mi15oh2qojb.bx.internal.cloudapp.net
nameserver 168.63.129.16

$ if /bin/grep $(hostname -I|cut -f1 -d" ") /etc/resolv.conf.dnsmasq; then echo "already replaced resolv.conf.dnsmasq"; else /bin/cp /etc/resolv.conf /etc/resolv.conf.dnsmasq; fi;
$ cat /etc/resolv.conf.dnsmasq 
# Generated by NetworkManager
search zkloo21cxfherh0mi15oh2qojb.bx.internal.cloudapp.net
nameserver 10.0.0.7

Reply:
Thanks, file existing is a much simpler signal.